### PR TITLE
feat(users): implement GET /users/me endpoint for authenticated user profile retrieval

### LINKED ISSUE #73: Implement /me endpoint for authenticated user profile and role context
### Description:
As a **user**,
I want to **retrieve my own profile information, role assignments, and organizational context**,
so that **I can see my details and understand what roles and entities I have access to in the system**.

### Acceptance Criteria:
- [ ] The system provides a `GET /me` endpoint accessible to all authenticated users
- [ ] The endpoint returns the authenticated user's complete profile including:
  - Basic user information (id, username, email, full_name, status)
  - Primary role and entity
  - All role assignments across different entities
  - Organizational context for each assignment
- [ ] The endpoint is protected by authentication but does not require admin role
- [ ] Users can only access their own profile (not other users' profiles)
- [ ] The response includes nested entity and role information
- [ ] The endpoint handles cases where user has no role assignments gracefully
- [ ] Unit tests verify authenticated users can access their profile
- [ ] E2E tests verify the endpoint works with real authentication tokens

### PR DESCRIPTION
**Pull request by JMB-7**
Key Changes

User Profile Endpoint Implementation

Added GET /users/me endpoint accessible to all authenticated users
Returns complete user profile including basic info, primary role/entity, and all role assignments
Provides organizational context through nested entity and role information
Handles users with no additional role assignments gracefully (empty array)

Controller Architecture Restructure

UsersController: handles public user endpoints at /users/* (JWT auth only)
AdminUsersController: manages admin operations at /admin/users/* (admin role required)
Separated concerns for better security and maintainability
Maintains backward compatibility for existing admin endpoints

Service Enhancement (UsersService)

findUserProfile(userId): retrieves user data with complete role assignment context
Leverages existing eager-loaded primary role and entity relationships
Fetches additional role assignments through UserRoleAssignment repository
Returns structured data with nested role and entity information for organizational context

Authentication Integration

Utilizes existing JWT authentication strategy and IdentityResolutionService
Resolves authenticated user through subject/issuer claims from JWT token
Ensures users can only access their own profile information (security isolation)
No admin privileges required - accessible to all authenticated users

Enhanced Features

Comprehensive user profile data including timestamps and status information
Complete role assignment details with entity hierarchy context
Graceful error handling for authentication failures
RESTful API design following existing patterns and conventions

Test Coverage

Added 3 comprehensive unit test cases for the /users/me endpoint
Tests successful profile retrieval with role assignments
Tests handling of users with no additional role assignments
Tests authentication error scenarios and edge cases
Updated existing admin controller tests to reflect architectural changes
All 92 tests pass with no breaking changes to existing functionality

### CODE REVIEW

File: backend/src/users/users.controller.ts [hunk 1]
```diff
@@ -1,15 +1,75 @@
...
+      })),
+    };
+  }
+}
 
 @Controller('admin/users')
 @UseGuards(AuthGuard('jwt'), RolesGuard)
 @Roles('admin')
-export class UsersController {
+export class AdminUsersController {
```
Comment (hndoss) [inline backend/src/users/users.controller.ts]: I think we can move this to another file for readability.

**Review by hndoss — APPROVED** [review]

**Review by hndoss — APPROVED** [review]
